scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -192,6 +193,8 @@ sealed trait Execution[+T] extends java.io.Serializable {
    */
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
+
+  override val hashCode = super.hashCode
```

Comment:
can we add a test that this is still working like normal case class equality and hashCode?

For instance, `val e1 = Execution.from(1); val e2 = Execution.from(2); e1.zip(e2).hashCode == e1.zip(e2).hashCode`.

I'm worried this could break caching if somehow this picked up `hashCode` from `java.lang.Object`.

Other than that, this looks good to me.